.github/workflows/build-and-test-3.yml

```diff
@@ -65,8 +65,57 @@ jobs:
           echo "::set-output name=version::$REVISION"
           echo "dubbo version: $REVISION"
 
+  unit-test-prepare:
+    name: " Preparation for Unit Test On ${{ matrix.os }} (JDK: ${{ matrix.jdk }})"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-18.04, windows-2019 ]
+        jdk: [ 8, 11 ]
+    env:
+      DISABLE_FILE_SYSTEM_TEST: true
+    steps:
+      - name: "Set up msys2 if necessary"
+        if: ${{ startsWith( matrix.os, 'windows') }}
+        uses: msys2/setup-msys2@v2
+        with:
+          release: false  # support cache, see https://github.com/msys2/setup-msys2#context
+      - name: "Download zookeeper binary archive in Linux OS"
+        if: ${{ startsWith( matrix.os, 'ubuntu') }}
+        run: |
+          mkdir -p ${{ github.workspace }}/.tmp/zookeeper/
+          wget -c https://archive.apache.org/dist/zookeeper/zookeeper-3.6.3/apache-zookeeper-3.6.3-bin.tar.gz -O ${{ github.workspace }}/.tmp/zookeeper/apache-zookeeper-bin.tar.gz ||
```

Comment:
It's better to check cache hit and only download if miss cache.  
See about https://github.com/actions/cache#example-workflow , some thing like: 
```yaml
    - name: Cache Primes
      id: cache-primes
      uses: actions/cache@v2
      with:
        path: prime-numbers
        key: ${{ runner.os }}-primes

    - name: Generate Prime Numbers
      if: steps.cache-primes.outputs.cache-hit != 'true'
      run: /generate-primes.sh -d prime-numbers
```

Reply:
The zookeeper server cache should load before download it, otherwise the cache may overwrite the cache file